### DISCUSSION TITLE
Provide error messages if math or latex syntax is incorrect

### DISCUSSION THREAD

**Discussion by venpopov** [quarto-dev/quarto-cli#8484] [Q&A] [answered]
### Description

I have an equation that was rendered just fine within visual editing mode of Rstudio:

![image](https://github.com/quarto-dev/quarto-cli/assets/12783882/7d357149-f952-4d64-9153-48406eb0da8a)

But when I rendered the html, it was not: 

![image](https://github.com/quarto-dev/quarto-cli/assets/12783882/403e6e72-0cdd-4c31-bc4d-82d1991a7db1)

I spend half an hour trying to solve until I decided to try rendering it to pdf and I got this error message:

![image](https://github.com/quarto-dev/quarto-cli/assets/12783882/97164441-bc87-48ae-af6c-757db258557f)

Removing the # symbol from the equation resolved the issue and it was rendered in the html output. But it would have saved me a lot of time if I got that error message during the html rendering, rather than just a weird output.

**Comment by mcanouil** [comment]
Unfortunately, we can't use screenshots.
Please share a small reproducible code and please indicate which version of Quarto and RStudio you are using.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

![A placeholder image](https://placehold.co/600x400.png)

The end.
````
`````

**Reply by venpopov** [reply]
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

$$
i = 1,…,5 \ \text{ (Task #)}
$$
````

**Comment by cderv** [comment, marked as answer]
> I have an equation that was rendered just fine within visual editing mode of Rstudio:
> (...)
> But when I rendered the html, it was not:

I think this comes from the fact that RStudio IDE may still use Mathjax 2.7 while Quarto HTML uses Mathjax 3 by default. 

Try this document 
````markdown
---
title: "Reproducible Quarto Document"
format: html
html-math-method: 
  method: mathjax
  url: "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/MathJax.js?config=TeX-AMS-MML_HTMLorMML"
---

$$
i = 1,…,5 \ \text{ (Task #)}
$$
````

This should show the equation correctly in the HTML

![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/2b58902c-2980-4d08-be4e-93b430321c14)

I think this is related upstream to 

* https://github.com/rstudio/rstudio/issues/8715
* https://github.com/rstudio/rstudio/issues/11535

Maybe you can open one specific to visual editor if none there already. 

> I spend half an hour trying to solve until I decided to try rendering it to pdf and I got this error message:

Using `#` in math without escaping in LaTeX is not valid. So this is why you get the error. 

Hope it helps understand

**Reply by venpopov** [reply]
Thanks for the explanation, it does help. My suggestion is to somehow raise an error like it happens when I tried rendering a pdf, so that the user knows what is the problem.
